src/flint/types/fmpz_poly.pyx

```diff
@@ -397,6 +397,59 @@ cdef class fmpz_poly(flint_poly):
         fmpz_poly_gcd(res.val, self.val, (<fmpz_poly>other).val)
         return res
 
+    def resultant(self, other):
+        """
+        Returns the resultant of *self* and *other*.
+
+            >>> A = fmpz_poly([1, 0, -1]); B = fmpz_poly([1, -1])
+            >>> A.resultant(B)
+            0
+            >>> C = fmpz_poly([1, 0, 0, 0, 0, -1, 1])
+            >>> D = fmpz_poly([1, 0, 0, -1, 0, 0, 1])
+            >>> C.resultant(D)
+            3
+            >>> f = fmpz_poly([1, -1] + [0] * 98 + [1])
+            >>> g = fmpz_poly([1] + [0] * 50 + [-1] + [0] * 48 + [1])
+            >>> f.resultant(g)
+            1125899906842623
+
+        """
+        cdef fmpz res
+        other = any_as_fmpz_poly(other)
+        if other is NotImplemented:
+            raise TypeError("cannot convert input to fmpz_poly")
+
+        res = fmpz.__new__(fmpz)
+        fmpz_poly_resultant(res.val, self.val, (<fmpz_poly>other).val)
+        return res
+
+    def resultant_modular(self, other):
+        """
+        Returns the resultant of *self* and *other* using Collins' 1971 modular
+        algorithm.
```

Comment:
I think it is better not to include this. There are lots of situations where FLINT provides variant functions for an operation like this and we don't generally wrap all of them in python-flint. It looks like the resultant method already switches between either using `fmpz_poly_resultant_modular` or `fmpz_poly_resultant_euclidean` depending on the degree and bitsize.